### ISSUE TITLE
Supporting for bulk suppressions

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3503]
## Summary

Request to add a bulk suppressions feature similar to [ESLint's Suppressions](https://eslint.org/docs/latest/use/suppressions) (introduced in ESLint v9.8.0).

## Motivation

When introducing new rules to an existing project, a large number of pre-existing violations are reported, making it impractical to enable the rule. Bulk suppressions allow teams to record and suppress existing violations while strictly enforcing the rule on new code.

## ESLint's Suppressions Feature Overview

- **`--suppress-all`**: Register all current violations into a suppression list
- **`--suppress-rule <rule>`**: Register only violations of a specific rule
- **`--prune-suppressions`**: Remove suppression entries for violations that have been fixed
- **`--suppressions-location <path>`**: Specify a custom path for the suppressions file
- **`eslint-suppressions.json`**: Manages suppression data as a JSON file (intended to be committed to the repository)
- Only "error" severity violations are suppressed ("warn" is not affected)

## Prior Art

This pattern is widely adopted across linters in various languages:

| Tool | Language | Feature Name | Matching Strategy |
|---|---|---|---|
| **PHPStan** | PHP | Baseline | Message regex + count + file path |
| **detekt** | Kotlin | Baseline | RuleID + finding signature (code structure) |
| **RuboCop** | Ruby | TODO | Per-rule file exclusion list |
| **ESLint** | JS/TS | Bulk Suppressions | File path + rule + count |
| **@rushstack/eslint-bulk** | JS/TS | Bulk Suppressions | File path + rule + scopeId (AST hierarchy) |
| **Stylelint** | CSS | Bulk Suppressions | Same as ESLint (count-based) |
| **Ruff** | Python | `--add-noqa` | Inline comment auto-insertion |
| **Biome** | JS/TS | Baseline | Under discussion ([Discussion #1064](https://github.com/biomejs/biome/discussions/1064)) |
| **oxlint** | JS/TS | Error baseline | Requested ([Discussion #1691](https://github.com/oxc-project/oxc/discussions/1691)) |

## Idea: Selector Scope

markuplint's strength is its powerful CSS selector engine. We could leverage this for more precise suppression matching by computing a **selector scope** for each suppression entry.

### Concept

When all violations of a rule in a file occur within a specific subtree, automatically compute the **Lowest Common Ancestor (LCA)** of the violation nodes and express it as a CSS selector:

```html
<nav>
  <ul>
    <li><a>...</a></li>  <!-- violation -->
    <li><a>...</a></li>  <!-- violation -->
    <li><a>...</a></li>  <!-- violation -->
  </ul>
</nav>
<footer>
  <a>...</a>  <!-- no violation -->
</footer>
```

LCA = `nav > ul` → Only violations within `nav > ul` are suppressed.

```json
{
  "filePath": "src/index.html",
  "ruleId": "some-rule",
  "scope": "nav > ul",
  "count": 3
}
```

### Design Considerations

- **Selector composition**: Tag names only (e.g., `nav > ul > li`) for stability; class/id would break on refactoring
- **Fallback**: When LCA resolves to `body` or `html`, omit the scope (suppress at file level)
- **Combined with count**: Scope alone is insufficient; use scope + count together for better precision
- **Phased approach**: Phase 1 with count-only (ESLint-compatible), Phase 2 adds selector scope

## Reference

- https://eslint.org/docs/latest/use/suppressions
- https://phpstan.org/user-guide/baseline
- https://detekt.dev/docs/introduction/baseline/
- https://developers.tiktok.com/blog/bulk-suppressions-a-new-eslint-feature-for-large-codebases
